Tools/CMake/FindAMReXFFTW.cmake

```diff
@@ -7,45 +7,142 @@ Finds the FFTW library.
 Imported Targets
 ^^^^^^^^^^^^^^^^
 
-This module provides the following imported target, if found:
+This module provides the following imported target:
 
-``FFTW``
+``AMReX::FFTW``
   The FFTW library
 
+Options/Control Variables
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``AMReX_FFTW_SEARCH``
+  FFTW search method (PKGCONFIG/CMAKE).
+  Defaults to CMake config packages on Windows and to PkgConfig pc files on Linux/macOS.
+
+``AMReX_FFTW_IGNORE_OMP``
+  Ignore FFTW3 OpenMP support, even if found.
+
 Result Variables
 ^^^^^^^^^^^^^^^^
 
 This will define the following variables:
 
 ``AMReXFFTW_FOUND``
-  True if the hypre library has been found.
-``FFTW_INCLUDES``
-  Include directories needed to use FFTW.
-``FFTW_LIBRARIES``
-  Libraries needed to link to FFTW.
+  True if the FFTW library has been found.
 
 This will also create an imported target, AMReX::FFTW.
+
 #]=======================================================================]
 
-if (NOT FFTW_INCLUDES)
-    find_path(FFTW_INCLUDES NAMES "fftw3.h" HINTS ${FFTW_ROOT}/include)
-endif()
+# Helper Functions ############################################################
+#
+option(AMReX_FFTW_IGNORE_OMP "Ignore FFTW3's OpenMP support, even if found" OFF)
+mark_as_advanced(AMReX_FFTW_IGNORE_OMP)
 
-if (NOT FFTW_LIBRARIES)
-    find_library(FFTW_LIBRARY NAMES "fftw3" HINTS ${FFTW_ROOT}/lib)
-    find_library(FFTWF_LIBRARY NAMES "fftw3f" HINTS ${FFTW_ROOT}/lib)
-    set(FFTW_LIBRARIES ${FFTW_LIBRARY} ${FFTWF_LIBRARY})
-endif()
+# Set the AMReX_FFTW_OMP=1 define on AMReX::FFTW if TRUE and print
+# a message
+#
+function(fftw_add_define HAS_FFTW_OMP_LIB)
+    if(HAS_FFTW_OMP_LIB)
+        message(STATUS "FFTW: Found OpenMP support")
+        target_compile_definitions(AMReX::FFTW INTERFACE AMReX_FFTW_OMP=1)
+    else()
+        message(STATUS "FFTW: Could NOT find OpenMP support")
+    endif()
+endfunction()
 
-include(FindPackageHandleStandardArgs)
+# Check if the found FFTW install location has an _omp library, e.g.,
+# libfftw3(f)_omp.(a|so) shipped and if yes, set the AMReX_FFTW_OMP=1 define.
+#
+function(fftw_check_omp library_paths fftw_precision_suffix)
+    find_library(HAS_FFTW_OMP_LIB fftw3${fftw_precision_suffix}_omp
```

Comment:
[find_library](https://cmake.org/cmake/help/latest/command/find_library.html) creates a cached variable, so currently it is only called once. Have to give it a unique name per precision.

Reply:
fixed